packages/astro/e2e/custom-client-directives.test.js

```diff
@@ -89,4 +89,13 @@ function testClientDirectivesShared() {
 		// Hydrated, this should be 1
 		await expect(counterValue).toHaveText('1');
 	});
+
+	test('Client directives should be passed options correctly', async ({ astro, page }) => {
```

Comment:
This doesn't test it for `client:visible` exactly... But still, it test that options are passed correctly to client directives (custom ones too, in this case)